src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicArray.cs

```diff
@@ -41,6 +41,10 @@ public class JsonDynamicArray : DynamicObject, IEnumerable<JsonNode?>
         }
     }
 
+    public bool Remove(JsonNode? item) => _jsonArray.Remove(item);
```

Comment:
[As I mentioned](https://github.com/OrchardCMS/OrchardCore/pull/15509#:~:text=The%20same%20applies%20to%20JsonDynamicArray%20as%20well.) Please update `_dictionary` in `JsonDynamicArray.Remove` as well.

Reply:
Updated